doc/bootloader/README.md

```diff
@@ -0,0 +1,113 @@
+
+The keywords here are *bootloader*, *BLE Stack* and *application firmware* :
+
+- A *bootloader* is a smaller firmware that is run when the device is started. Its most common tasks are to initialize the hardware and load the actual firmware. It can also be involved in the firmware update procedure.
+- A *BLE Stack* is a software library that implements the software functionalities needed to support BLE.
+- Finally, the *application firmware* is the actual firmware like InfiniTime.
+
+## The design
+
+Our goals when we designed the bootloader and the OTA procedure for the PineTime were:
+
+- to support multiple application firmwares;
+- to give as much freedom to firmware developers as possible;
+- to be as safe and reliable as possible.
+
+When designing the OTA procedure for the PineTime, we basically had 2 options : implement the OTA procedure in the bootloader or in the application firmware.
+
+Using the bootloader to handle the OTA is tempting as this would allow to implement it only once, and the application firmware would not need to care about it. However, the downside is that the bootloader would need to integrate the whole BLE stack.
+
+Indeed, the BLE stack is quite heavy in flash memory and a lot of memory would be waster if both the bootloader **and** the application firmware would integrate their own BLE stack. Of course, both the bootloader and the application firmware could share the same stack, but that's against our "freedom to the developers" goal : firmware developer would be forced to use whatever BLE stack we used in the bootloader.
+
+You've probably guessed it, we chose the 2nd solution, and we decided to implement the OTA procedure in the application firmware. In this case, the bootloader is much simpler, and firmware developers have a lot more freedom in their design choices (like the BLE stack to integrate, the data transfer protocol to implement,...).
+
+Here is basically how it works:
+
+- the application firmware exposes a BLE "firmware update" service. In InfiniTime we decided to implement the **nRF DFU** protocol.
+- this update service stores the new firmware in a temporary location in the *external flash memory* and checks the integrity of the data (CRC).
+- when the new firmware is completely transferred and checked, the watch is rebooted and the bootloader is ran again. The bootloader detects that a new firmware is available in the *external* flash memory and safely swaps the current firmware in the *internal* memory with the new one from the *external* memory.
+- When the swapping is done and checked, the new firmware is executed from the *internal* memory.
+- If the new firmware does not properly run, the bootloader will automatically revert to the older firmware by swapping again firmwares from the external to the internal memory.
+
+This procedure involves multiple software modules and rely on some hardware features. Let's go into more details!
```

Comment:
I suggest one line per sentence, less messy diffs

```suggestion
This procedure involves multiple software modules and rely on some hardware features.
Let's go into more details!
```

I love the enthusiasm by the way!